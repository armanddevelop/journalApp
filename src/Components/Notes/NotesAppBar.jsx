import { useDispatch, useSelector } from "react-redux";
import {
  startSaveNoteAction,
  startUploadImageAction,
} from "../../Actions/notes";

export const NotesAppBar = () => {
  const dispatch = useDispatch();
  const { active: note } = useSelector(({ notes }) => notes);
  const handleSaveNote = () => {
    dispatch(startSaveNoteAction(note));
  };
  const handlePictureClick = () => {
    document.getElementById("fileSelector").click();
  };
  const handleFileChange = ({ target }) => {
    const file = target.files[0];
    if (!file) return;
    dispatch(startUploadImageAction(file));
    document.getElementById("fileSelector").value = "";
  };

  return (
    <div className="notes__appbar">
      <span>19/03/2022</span>
      <input
        type="file"
        id="fileSelector"
        name="picture"
        style={{ display: "none" }}
        onChange={(e) => handleFileChange(e)}
      />
      <div>
        <button className="btn" onClick={handlePictureClick}>
          Picture
        </button>
        <button className="btn" onClick={handleSaveNote}>
          Save
        </button>
      </div>
    </div>
  );
};
