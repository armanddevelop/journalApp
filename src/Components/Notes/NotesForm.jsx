import { useEffect, useRef } from "react";
import { useSelector, useDispatch } from "react-redux";
import { notesActiveAction, startDeleteNoteAction } from "../../Actions/notes";
import { useForm } from "../../Hooks/useForm";

export const NotesForm = () => {
  const { active: note } = useSelector(({ notes }) => notes);
  const dispatch = useDispatch();
  const { values, handleInputChange, reset } = useForm(note);
  const { title, body } = values;
  const idActive = useRef(note.id);

  useEffect(() => {
    if (idActive.current !== note.id) {
      reset(note);
      idActive.current = note.id;
    }
  }, [note, reset]);

  useEffect(() => {
    const { id, date, title, body, url } = values;
    const noteObj = {
      date,
      title,
      body,
      url,
    };
    dispatch(notesActiveAction(id, noteObj));
  }, [values, dispatch]);

  const handleDelete = () => {
    dispatch(startDeleteNoteAction(idActive.current));
  };

  return (
    <div className="notes__content">
      <input
        type="text"
        placeholder="Some awsome title"
        className="notes__title-input"
        autoComplete="off"
        autoFocus={true}
        name="title"
        value={title}
        onChange={(e) => handleInputChange(e)}
      />
      <textarea
        placeholder="What are you doing today"
        className="notes__textarea"
        name="body"
        value={body}
        onChange={(e) => handleInputChange(e)}
      ></textarea>
      <button className="btn btn-danger" onClick={handleDelete}>
        Delete note
      </button>
      {note.url && (
        <div className="notes__image">
          <img src={note.url} alt="source" />
        </div>
      )}
    </div>
  );
};
