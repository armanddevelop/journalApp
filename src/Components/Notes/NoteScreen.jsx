import { NotesAppBar } from "./NotesAppBar";

export const NoteScreen = () => {
  return (
    <div className="notes__main-content">
      <NotesAppBar />
      <div className="notes__content">
        <input
          type="text"
          placeholder="Some awsome title"
          className="notes__title-input"
          autoComplete="off"
        />
        <textarea
          placeholder="What are you doing today"
          className="notes__textarea"
        ></textarea>
        <div className="notes__image">
          <img
            src="https://assets.leevalley.com/Size5/10061/49L0795-everyman-s-journal-u-02-r.jpg"
            alt="source"
          />
        </div>
      </div>
    </div>
  );
};
