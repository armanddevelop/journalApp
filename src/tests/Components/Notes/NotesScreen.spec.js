import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { NoteScreen } from "../../../Components/Notes/NoteScreen";

const initState = {
  auth: {
    name: "Alicha",
  },
  ui: {
    errorMsgFb: {
      error: false,
      msgError: "",
    },
    loading: false,
  },
  notes: {
    active: {
      id: "safdbdg32435",
    },
    note: {},
    notes: [],
    saveNote: {
      isSave: true,
      message: "All rigth",
      title: "note save",
    },
  },
};
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
let store = mockStore(initState);
store.dispatch = jest.fn();

describe("Test in NotesScreen component", () => {
  const wrapper = mount(
    <Provider store={store}>
      <NoteScreen />
    </Provider>
  );

  it("SnapShot to NotesScreen component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render NotesForm component", () => {
    expect(wrapper.find(".notes__content").exists()).toEqual(true);
    expect(wrapper.find(".loading").exists()).toEqual(false);
  });

  it("Should render loading component", () => {
    const initState = {
      auth: {
        name: "Alicha",
      },
      ui: {
        errorMsgFb: {
          error: false,
          msgError: "",
        },
        loading: true,
      },
      notes: {
        active: {
          id: "safdbdg32435",
        },
        note: {},
        notes: [],
        saveNote: {
          isSave: true,
          message: "All rigth",
          title: "note save",
        },
      },
    };
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);
    let store = mockStore(initState);
    const wrapper = mount(
      <Provider store={store}>
        <NoteScreen />
      </Provider>
    );

    expect(wrapper.find(".notes__content").exists()).toEqual(false);
    expect(wrapper.find(".loading").exists()).toEqual(true);
  });
});
