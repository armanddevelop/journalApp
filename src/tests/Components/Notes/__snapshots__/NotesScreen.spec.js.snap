// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in NotesScreen component SnapShot to NotesScreen component 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      19/03/2022
    </span>
    <input
      id="fileSelector"
      name="picture"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div>
    <Portal
      containerInfo={
        <body
          style="padding-right: 1024px; overflow: hidden;"
        >
          <div
            aria-describedby="transition-modal-description"
            aria-labelledby="transition-modal-title"
            class="MuiModal-root css-79ws1d-MuiModal-root"
            role="presentation"
          >
            <div
              aria-hidden="true"
              class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
              style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            />
            <div
              data-test="sentinelStart"
              tabindex="0"
            />
            <div
              class="notes__modal MuiBox-root css-0"
              style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
              tabindex="-1"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-outlinedSuccess MuiAlert-outlined css-dcrfqy-MuiPaper-root-MuiAlert-root"
                role="alert"
              >
                <div
                  class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="SuccessOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiAlert-message css-acap47-MuiAlert-message"
                >
                  <div
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
                    severity="success"
                  >
                    <strong>
                      note save
                    </strong>
                  </div>
                  All rigth
                </div>
              </div>
            </div>
            <div
              data-test="sentinelEnd"
              tabindex="0"
            />
          </div>
        </body>
      }
    >
      Array [
        "",
        <div
          aria-describedby="transition-modal-description"
          aria-labelledby="transition-modal-title"
          className="MuiModal-root css-79ws1d-MuiModal-root"
          onKeyDown={[Function]}
          role="presentation"
        >
          Array [
            "",
            Array [
              "",
              <div
                aria-hidden={true}
                className="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
                onClick={[Function]}
                style={
                  Object {
                    "opacity": 1,
                    "visibility": undefined,
                  }
                }
              />,
            ],
          ]
          Array [
            <div
              data-test="sentinelStart"
              onFocus={[Function]}
              tabIndex={0}
            />,
            Array [
              "",
              <div
                className="notes__modal MuiBox-root css-0"
                onFocus={[Function]}
                style={
                  Object {
                    "opacity": 1,
                    "visibility": undefined,
                  }
                }
                tabIndex="-1"
              >
                Array [
                  "",
                  Array [
                    "",
                    <div
                      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-outlinedSuccess MuiAlert-outlined css-dcrfqy-MuiPaper-root-MuiAlert-root"
                      role="alert"
                    >
                      Array [
                        "",
                        <div
                          className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
                        >
                          Array [
                            "",
                            <svg
                              aria-hidden={true}
                              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                              data-testid="SuccessOutlinedIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
                              />
                            </svg>,
                          ]
                        </div>,
                      ]
                      Array [
                        "",
                        <div
                          className="MuiAlert-message css-acap47-MuiAlert-message"
                        >
                          Array [
                            "",
                            Array [
                              "",
                              <div
                                className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
                                severity="success"
                              >
                                <strong>
                                  note save
                                </strong>
                              </div>,
                            ],
                          ]
                          All rigth
                        </div>,
                      ]
                    </div>,
                  ],
                ]
              </div>,
            ],
            <div
              data-test="sentinelEnd"
              onFocus={[Function]}
              tabIndex={0}
            />,
          ]
        </div>,
      ]
    </Portal>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      autoFocus={true}
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awsome title"
      type="text"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What are you doing today"
    />
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Delete note
    </button>
  </div>
</div>
`;
